### PR TITLE
fix: release process automation

### PR DESCRIPTION
**Pull request by kuannie1**
changes:
* deleting the duplicate/nonfunctional goreleaser workflows in .github/workflows
* Adding an official "release" workflow
* copying the release & prerelease process from the `happy` project 
* conventional commit check

### CODE REVIEW

File: .goreleaser.prerelease.yml [hunk 1]
```diff
@@ -13,32 +9,21 @@ builds:
...
-  - vendor: Chan Zuckerberg Initiative, LLC
-    license: MIT
-    description: AWS cli single sign on using OIDC
-    formats:
-      - deb
-
-archives:
-  - files:
-      - none*
+      - "-w -s -X github.com/chanzuckerberg/aws-oidc/pkg/util.ReleaseGitSha={{.Commit}} -X github.com/chanzuckerberg/aws-oidc/pkg/util.ReleaseVersion={{.Version}}"
```
Comment (edulop91) [inline .goreleaser.prerelease.yml:12]: these should match those on the left

File: .goreleaser.prerelease.yml [hunk 2]
```diff
@@ -13,32 +9,21 @@ builds:
       - amd64
       - arm64
     ldflags:
-      - '-w -s -X github.com/chanzuckerberg/aws-oidc/pkg/util.GitSha={{.Commit}} -X github.com/chanzuckerberg/aws-oidc/pkg/util.Version={{.Version}} -X github.com/chanzuckerberg/aws-oidc/pkg/util.Dirty=false -X github.com/chanzuckerberg/aws-oidc/pkg/util.Release=true'
-
-dockers:
```
Comment (edulop91) [inline .goreleaser.prerelease.yml:18]: aws-oidc has a docker server-side component so we'll need to bring this back in a subsequent change

File: .goreleaser.yml [hunk 3]
```diff
@@ -13,22 +9,10 @@ builds:
...
-#   - dockerfile: Dockerfile
-#     image_templates:
-#       - docker.pkg.github.com/chanzuckerberg/aws-oidc/aws-oidc:v{{.Version}}
-#     extra_files:
-#       - cmd
-#       - pkg
-#       - go.mod
-#       - go.sum
-#       - main.go
+      - "-w -s -X github.com/chanzuckerberg/aws-oidc/pkg/util.ReleaseGitSha={{.Commit}} -X github.com/chanzuckerberg/aws-oidc/pkg/util.ReleaseVersion={{.Version}}"
```
Comment (edulop91) [inline .goreleaser.yml:12]: same here

File: .goreleaser.prerelease.yml (hunk 1 above)
Comment (kuannie1) [inline .goreleaser.prerelease.yml:12]:
```suggestion
      - '-w -s -X github.com/chanzuckerberg/aws-oidc/pkg/util.GitSha={{.Commit}} -X github.com/chanzuckerberg/aws-oidc/pkg/util.Version={{.Version}} -X github.com/chanzuckerberg/aws-oidc/pkg/util.Dirty=false -X github.com/chanzuckerberg/aws-oidc/pkg/util.Release=true'
```

File: .goreleaser.yml (hunk 3 above)
Comment (kuannie1) [inline .goreleaser.yml:12]:
```suggestion
      - "-w -s -X github.com/chanzuckerberg/aws-oidc/pkg/util.GitSha={{.Commit}} -X github.com/chanzuckerberg/aws-oidc/pkg/util.Version={{.Version}} -X github.com/chanzuckerberg/aws-oidc/pkg/util.Dirty=false -X github.com/chanzuckerberg/aws-oidc/pkg/util.Release=true"
```

File: .goreleaser.prerelease.yml (hunk 2 above)
Comment (kuannie1) [reply .goreleaser.prerelease.yml:18]: thanks for the reminder! completely forgot about that. I just added it back

File: .goreleaser.prerelease.yml [hunk 4]
```diff
@@ -14,6 +10,8 @@ builds:
       - arm64
     ldflags:
       - '-w -s -X github.com/chanzuckerberg/aws-oidc/pkg/util.GitSha={{.Commit}} -X github.com/chanzuckerberg/aws-oidc/pkg/util.Version={{.Version}} -X github.com/chanzuckerberg/aws-oidc/pkg/util.Dirty=false -X github.com/chanzuckerberg/aws-oidc/pkg/util.Release=true'
+    flags:
+      - -trimpath
```
Comment (kuannie1) [inline .goreleaser.prerelease.yml:14]:
```suggestion
```

File: .github/workflows/conventional_commits_title.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      - edited
+      - opened
+      - synchronize
+      - reopened
+
+jobs:
+  conventional_commit_title:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v5
```
Comment (dtsai-czi) [inline .github/workflows/conventional_commits_title.yml:14]: You can pin this script to our github actions repo too like in here: https://github.com/chanzuckerberg/shared-infra/blob/main/.github/workflows/conventional_commits_title.yml#L14

Comment (kuannie1) [reply .github/workflows/conventional_commits_title.yml]: thank you so much! Yeah, I think that's a good recommendation. I appreciate it

**Review by alexlokshin-czi — APPROVED** [review]

**Pull request merged**